app.rb

```diff
@@ -0,0 +1,156 @@
+require './book'
+require './person'
+require './student'
+require './teacher'
+require './rental'
+
+class SchoolApp
+  def initialize
+    @books = []
+    @people = []
+    @rentals = []
+  end
+
+  def welcome
+    puts 'Welcome to School Library App!'
+  end
+
+  def menu
+    puts 'Please choose an option by entering a number: '
+    puts '1 - List all books'
+    puts '2 - List all people'
+    puts '3 - Create a person'
+    puts '4 - Create a book'
+    puts '5 - Create a rental'
+    puts '6 - List all rentals for a given person id'
+    puts '7 - Exit'
+  end
+
+  def list_of_books
+    if @books.empty?
+      puts 'There is no books in the list'
+    else
+      @books.each { |book| puts "Title: \"#{book.title}\", Author: #{book.author}" }
+    end
+  end
+
+  def person_display(person)
+    print "[#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+    puts ", Specialization: #{person.specialization}" if person.instance_of?(Teacher)
+    puts ", Classroom: #{person.classroom}" if person.instance_of?(Student)
+  end
+
+  def list_of_people
+    if @people.empty?
+      puts 'There is no people found'
+    else
+      @people.each { |person| person_display(person) }
+    end
+  end
+
+  def add_person_info
+    print 'Do you want to create a student (1) or a teacher (2)? [Input the number]: '
+    person_type = gets.chomp
+
+    print 'Age: '
+    age = gets.chomp
+
+    print 'Name: '
+    name = gets.chomp
+
+    create_person(person_type, age, name)
+  end
+
+  def create_person(person_type, age, name)
+    case person_type
+    when '1'
+      print 'Has parent permission? [Y/N]: '
+      parent_permission = gets.chomp.downcase == 'y'
+      print 'What is the student\'s classroom? '
+      classroom = gets.chomp
+      @people.push(Student.new(age, name, classroom, parent_permission: parent_permission))
+    when '2'
+      print 'Specialization: '
+      specialization = gets.chomp
+      @people.push(Teacher.new(age, name, specialization))
+    else
+      puts 'Invalid option'
+      add_person_info
+    end
+
+    puts 'Person was created successfully'
+  end
+
+  def book_creation
+    print 'Title: '
+    title = gets.chomp
+
+    print 'Author: '
+    author = gets.chomp
+
+    @books.push(Book.new(title, author))
+    puts 'Book created successfully'
+  end
+
+  def rental_creation
+    puts 'Select a book from the following list by name'
```

Comment:
- [ ] Please change this line ending with 'number' instead of 'name' as we only demand the index of the book.